contrib/nydusify/pkg/converter/provider/provider.go

```diff
@@ -152,6 +156,36 @@ func (pvd *Provider) Push(ctx context.Context, desc ocispec.Descriptor, ref stri
 	return push(ctx, pvd.store, rc, desc, ref)
 }
 
+func (pvd *Provider) Import(ctx context.Context, reader io.Reader) (string, error) {
+	iopts := importOpts{
+		dgstRefT: func(dgst digest.Digest) string {
+			return "nydus" + "@" + dgst.String()
+		},
+		skipDgstRef:     func(name string) bool { return name != "" },
+		platformMatcher: pvd.platformMC,
+	}
+	images, err := load(ctx, reader, pvd.store, iopts)
+	if err != nil {
+		return "", err
+	}
+
+	if len(images) != 1 {
+		return "", errors.New("importing multiple images")
```

Comment:
We should support multi platforms image.

Reply:
![image](https://github.com/user-attachments/assets/6a9573e4-e41b-4aba-91c5-ebc048f5d20a)

The length of `images` array represents the number of images in the tarball, not manifests. As shown in the figure, there are two images `airpine` and `nginx` in the tarball, each of which is multi platform image storing multiple manifests(`linux/amd64` and `linux/arm64/v8`).

Since all of our exported tarball will only contain 1 image and it's manifests, it can be assumed that if the length of `images` in the tarball is greater than 1, the tarball is not formatted correctly. 

So maybe this code doesn't need to be changed?

Reply:
Thanks for the detail, LGTM.